Automattic/jetpack
# Connection: display errors while connecting user account

#### Changes proposed in this Pull Request:
* If user tries to connect in site-only mode and connection fails, display an error message.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_Before checking out the branch, try to reproduce the existing bug:_
1. Connect in site-only mode.
2. Go back to Jetpack Dashboard and initiate the user connection.
3. When you get to the connection screen, switch your browser to the offline mode, or disconnect from the internet altogether.
4. Click "Connect your user account". The button should get stuck in the "loading" mode, no error message should show up.
5. Disable the offline mode (or reconnect to the internet).

_Now checkout this branch, and rebuild jetpack if needed._

6. Reload the "Connect your user account" screen.
7. Switch back to the offline mode, and click the button. The loader should flicker and quickly disappear. The error message should appear.
8. Disable the offline mode, and try to connect again without reloading the "errored" page. Confirm that user connection goes through.

<img width="299" alt="Screen Shot 2022-01-12 at 4 29 33 PM" src="https://user-images.githubusercontent.com/1341249/149224321-673bf25a-f40f-4669-936d-94e4f82ce500.png">
